src/main/java/randoop/condition/SpecificationCollection.java

```diff
@@ -0,0 +1,312 @@
+package randoop.condition;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import randoop.compile.SequenceClassLoader;
+import randoop.compile.SequenceCompiler;
+import randoop.condition.specification.OperationSignature;
+import randoop.condition.specification.OperationSpecification;
+import randoop.reflection.TypeNames;
+import randoop.util.Log;
+import randoop.util.MultiMap;
+
+/**
+ * A collection of {@link OperationSpecification} objects. Also a map from an {@link
```

Comment:
How are these two parts of the data structure related?

Also say how this is intended to be used.  Does it represent all the specifications for all methods in an entire project?